pandas/tests/test_categorical.py

```diff
@@ -3625,6 +3625,62 @@ def test_cat_accessor_api(self):
             invalid.cat
         self.assertFalse(hasattr(invalid, 'cat'))
 
+    def test_str_accessor_api_for_categorical(self):
+        # https://github.com/pydata/pandas/issues/10661
+        from pandas.core.strings import StringMethods
+        s = Series(list('aabb'))
+        c = s.astype('category')
+        self.assertIsInstance(c.str, StringMethods)
+
+        res = c.str.contains("a")
+        exp = s.str.contains("a")
+        tm.assert_series_equal(res, exp)
+
+        res = c.str.upper()
+        exp = s.str.upper()
+        tm.assert_series_equal(res, exp)
+
+        res = c.str.len()
+        exp = s.str.len()
+        tm.assert_series_equal(res, exp)
+
+        res = c.str.findall("a")
+        exp = s.str.findall("a")
+        tm.assert_series_equal(res, exp)
+
+        s = s+" "+s
+        c = s.astype('category')
+        res = c.str.split(" ")
+        exp = s.str.split(" ")
+        tm.assert_series_equal(res, exp)
+
+        res = c.str.split(" ", expand=True)
+        exp = s.str.split(" ", expand=True)
+        tm.assert_frame_equal(res, exp)
+
+        invalid = Series([1,2,3]).astype('category')
+        with tm.assertRaisesRegexp(AttributeError, "Can only use .str accessor with string"):
+            invalid.str
+        self.assertFalse(hasattr(invalid, 'str'))
+
+    def test_dt_accessor_api_for_categorical(self):
+        # https://github.com/pydata/pandas/issues/10661
+        from pandas.tseries.common import Properties
+        from pandas.tseries.index import date_range
+        s = Series(date_range('1/1/2015', periods=5))
+
```

Comment:
same here: IMO that thing is tested:

dt works on Index, so that path is covered and we basically do this here: take the categories, which are an index, do the dt things on it and the only difference to a dt on an index is, that it uses `take1d`on the result.